homeassistant/components/pyload/button.py

```diff
@@ -80,4 +83,10 @@ class PyLoadBinarySensor(BasePyLoadEntity, ButtonEntity):
 
     async def async_press(self) -> None:
         """Handle the button press."""
-        await self.entity_description.press_fn(self.coordinator.pyload)
+        try:
+            await self.entity_description.press_fn(self.coordinator.pyload)
+        except (CannotConnect, InvalidAuth, OSError, ClientConnectorError) as e:
```

Comment:
In the description every is resolved as a connection error. Would be useful to specify and hint out more to the users what the actual reason is.

Reply:
These are the 2 exceptions that can raise at the moment, more details would mean either printing out the stacktrace or modifying the library.

Reply:
Well they usually contain a string of extra data. Which you could add as a replaced string in the error message.

Reply:
My point being, if there's an invalid authentication raised, the user would be helped if they know. Now it's all considered, in the description, as a connection error. Making it dynamic to specify more which exception is raised, helps users to resolve it more themselves. :)

Reply:
Argh, just noticed, OSError and ClientConnectError shouldn't be there, they are already catched by CannotConnect

Reply:
> Well they usually contain a string of extra data. Which you could add as a replaced string in the error message.

That would draw having translated error messages totally unnecessary

Reply:
Shouldn't `InvalidAuth` trigger re-auth instead?

Reply:
Printing out the exception message would result in something like i had before, with the difference it will be not translated:
```
Unable to send command to pyLoad due to a connection error, try again later: Restarting all failed files failed due to request exception
```
`Restarting all failed files failed due to request exception``is the message text from the exception

Reply:
> Shouldn't `InvalidAuth` trigger re-auth instead?

This as well.

Reply:
> Shouldn't `InvalidAuth` trigger re-auth instead?

The DataUpdateCoordinator will handle ConfigEntryAuthError, i think at this point it won't be catched.

Reply:
yepp, as presumed, `ConfigEntryAuthFailed` is not catched at this point  and will be shown in the log as unexcepcted exception

Reply:
> Printing out the exception message would result in something like i had before, with the difference it will be not translated:
> ```
> Unable to send command to pyLoad due to a connection error, try again later: Restarting all failed files failed due to request exception
> ```
> `Restarting all failed files failed due to request exception``is the message text from the exception 
> 

Then don't print, but try to help out the user to describe what's going on and potentially what they can do to fix it.

Reply:
> > Printing out the exception message would result in something like i had before, with the difference it will be not translated:
> > ```
> > Unable to send command to pyLoad due to a connection error, try again later: Restarting all failed files failed due to request exception
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > `Restarting all failed files failed due to request exception``is the message text from the exception
> 
> Then don't print, but try to help out the user to describe what's going on and potentially what they can do to fix it.

Like for example to try again later? At this point, usually only temporary hiccups can happen or if it is a permanent problem, it will only be possible to trigger in the 20 seconds before the UpdateCoordinator renders all entities unavailable.

Reply:
Your scope of exceptions is changed in the due course. If InvalidAuth remains in, it would be good to point out it due to authentication. Else, if it's now only reduced to ConnectionError only, then it no longer suffices to be more descriptive. :)

Reply:
sorry for the confusion, these were artifacts from when i had problems, because none of the raised exceptions were catched with HomeAssistantError. (Only ServerValidationError is catched here)